### PR TITLE
Searching

### LINKED ISSUE #2725: Search is not working in the order details data table from the admin panel.
image

client panel->my-orders->dowloads
search and sorting not working
Screenshot from 2023-12-21 11-24-07

### PR DESCRIPTION
**Pull request by Gurmeenkour99**
Fixes #2725 

1.Fix Searching issue in client panel>>my_orders>>download>>product_list and Admin Panel>>users>>orders_list
2.Fix the order of versions in client panel>>my_orders>>download>>product_list .Latest versions are at the top .

### CODE REVIEW

File: app/Http/Controllers/User/AdminOrderInvoiceController.php [hunk 1]
```diff
@@ -209,6 +210,15 @@ class='btn btn-sm btn-secondary btn-xs'".tooltip('View')."<i class='fa fa-eye'
                             style='color:white;'> </i></a>";
                         })
                         ->rawColumns(['checkbox', 'date', 'product', 'number', 'version', 'expiry', 'status', 'action'])
+                        ->filter(function ($query) use ($request) {
+                            if ($request->has('search') && ! empty($request->input('search')['value'])) {
+                                $searchValue = $request->input('search')['value'];
+                                $query->where(function ($subquery) use ($searchValue) {
+                                    $subquery->where('orders.number', 'like', "%{$searchValue}%")
+                                        ->orWhere('products.name', 'like', "%{$searchValue}%");
+                                });
```
Comment (sowmiyasubramaniyan) [inline app/Http/Controllers/User/AdminOrderInvoiceController.php:219]: what about version and status search?

**Review by sowmiyasubramaniyan — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/User/AdminOrderInvoiceController.php (hunk 1 above)
Comment (Gurmeenkour99) [reply app/Http/Controllers/User/AdminOrderInvoiceController.php]: done

**Review by sowmiyasubramaniyan — APPROVED** [review]

**Review by sandesh556 — APPROVED** [review]

**Comment by Sabaric12** [comment]
Round 1: Testing Summary
Tested this PR with the existing instance in the local
Now the mentioned issues are working fine we can able to search for the product with the title and the description in the client panel my order was downloaded in the product list and in the admin panel users' order details with the order no product and status able to  search
Total time in days: 2 hours
Total issues: 0
Minor - 0 
Major - 0 
Critical - 0
No of times PR was sent for Functionality correction:0
PR breaks existing functionality: No

**Review by Sabaric12 — APPROVED** [review]

**Review by SarfrazS — APPROVED** [review]
Pointed issues are fixed now searching properly.
image

image

**Pull request merged**